### ISSUE TITLE
Using freeze can lead to fractional second differences

### ISSUE THREAD

**Issue by aprescott** [travisjeffery/timecop#75]
I've encountered this bug before and haven't been able to work out what's causing it, and unfortunately don't have a minimal test case that reproduces it. I upgraded to 0.6.1 and the problem is still there.

I have the following in an RSpec test case:

``` ruby
old_update_time = @temporary_post.updated
t = Time.now + 50

Timecop.freeze(t) do
  @temporary_post.update!
  @temporary_post.updated.should_not == old_update_time
  @temporary_post.updated.utc.should == t.utc # this fails
  @temporary_post.headers[:updated].should == t
end
```

`update!` here just sets `updated` to be `Time.now`:

``` ruby
def update!
  # headers is just a Hash
  @source.headers.delete(:update)
  set_updated_time(Time.now)
  # this writes out to disk
  save
end

# ...

def set_updated_time(time)
  @source.headers[:updated] = time.xmlschema
  headers_changed!
end
```

Running this test as part of the suite gives me this failure:

```
     Failure/Error: @temporary_post.updated.utc.should == t.utc
       expected: 2013-03-18 17:21:22 UTC
            got: 2013-03-18 17:21:22 UTC (using ==)
```

Obviously, the two times appear to be the same. If you check different offsets in different times, the problem shows itself:

``` ruby
old_update_time = @temporary_post.updated
t = Time.now + 50

Timecop.freeze(t) do
  puts Time.now - t # 0.0
  sleep 2

  @temporary_post.update!
  @temporary_post.updated.should_not == old_update_time

  puts Time.now - t # 0.0
  puts @temporary_post.updated.utc - Time.now # -0.74104631
  puts @temporary_post.updated.utc - t        # -0.74104631
  sleep 2

  @temporary_post.updated.utc.should == t.utc
  @temporary_post.headers[:updated].should == t
end
```

Inline comments added showing the output, but:

```
0.0
0.0
-0.74104631
-0.74104631
```

Basically, it seems that by the time `Time.now` is called in `update!`, Timecop hasn't finished "settling" the time? So `update!`'s `Time.now` is **behind** the `freeze` block by 0.74104631 seconds (on my machine), yet in the call site `freeze` block itself, `Time.now` is indeed `t`.

Unfortunately this minimal example **doesn't** reproduce it:

``` ruby
require "timecop"

def do_some_stuff
  Time.now
end

t = Time.now + 50
Timecop.freeze(t) do
  raise "argh!" unless Time.now == t
  raise "Times don't match" unless Time.now == do_some_stuff
end
```
